### PR TITLE
Add price events manager

### PR DESCRIPTION
**Pull request by Herklos**
Done in PR : 
- Add price events manager implementation
- Add recent trades manager tests
- Add price events manager tests

### CODE REVIEW

**Comment by Herklos** [comment]
To be done in the next PR : 
- Integrate to Order check
- Remove old Order check mechanism
- Add trailing stop Order

File: octobot_trading/data_manager/price_events_manager.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+from asyncio import Event
+
+from octobot_commons.logging.logging_util import get_logger
+from octobot_trading.enums import ExchangeConstantsOrderColumns as ECOC
+
+
+class PriceEventsManager:
+    """
+    Manage price events for a specific price and timestamp
+    Mainly used for updating Order status
```
Comment (GuillaumeDSM) [inline octobot_trading/data_manager/price_events_manager.py:25]: 👍

File: octobot_trading/data_manager/price_events_manager.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        Remove the event from events list
+        :param event_to_remove: the event to remove
+        """
+        for price_event_data in self.events:
+            if event_to_remove in price_event_data:
+                return self.events.remove(price_event_data)
+
+    def _check_events(self, price, timestamp):
+        """
+        Check each price, timestamp pair event should be triggered
```
Comment (GuillaumeDSM) [inline octobot_trading/data_manager/price_events_manager.py:104]: I this there are missing words :) 
Check **for** each price, timestamp pair event **if it** should be triggered

File: octobot_trading/data_manager/price_events_manager.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        price_event_tuple = _new_price_event(price, timestamp, trigger_above)
+        self.events.append(price_event_tuple)
+        return price_event_tuple[PriceEventsManager.PRICE_EVENT_INDEX]
+
+    def remove_event(self, event_to_remove):
+        """
+        Public method that calls _remove_event()
+        :param event_to_remove: the event to remove
+        """
+        return self._remove_event(event_to_remove)
```
Comment (GuillaumeDSM) [inline octobot_trading/data_manager/price_events_manager.py:82]: sad that we had to do this for cython :/

File: tests/data_manager/test_price_events_manager.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                                 timestamp=random_timestamp(max_value=random_timestamp_1 - 1))])
+        with pytest.raises(AssertionError):
+            price_event_1_set.assert_called_once()
+        price_events_manager.handle_recent_trades(
+            [random_recent_trade(price=random_price(max_value=random_price_1 - 1)),
+             random_recent_trade(price=random_price_1,
+                                 timestamp=random_timestamp_1),
+             random_recent_trade(price=random_price(max_value=random_price_1 - 1)),
+             random_recent_trade(price=random_price(max_value=random_price_1 - 1))])
+        price_event_1_set.assert_called_once()
```
Comment (GuillaumeDSM) [inline tests/data_manager/test_price_events_manager.py:70]: 👍

File: tests/data_manager/test_price_events_manager.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    if not os.getenv('CYTHON_IGNORE'):
+        assert price_events_manager.events
+        price_events_manager.remove_event(event_1)
+        assert event_1 not in price_events_manager.events
+        assert len(price_events_manager.events) == 1
+        price_events_manager.remove_event(Event())
+        assert len(price_events_manager.events) == 1
+        price_events_manager.remove_event(event_2)
+        assert event_2 not in price_events_manager.events
+        assert len(price_events_manager.events) == 0
```
Comment (GuillaumeDSM) [inline tests/data_manager/test_price_events_manager.py:165]: 💯

File: tests/data_manager/test_prices_manager.py [hunk 6]
```diff
@@ -34,21 +36,20 @@ async def test_initialize(backtesting_exchange_manager):
...
 
 
-async def test_set_mark_price(backtesting_exchange_manager):
-    prices_manager = PricesManager(backtesting_exchange_manager)
+async def test_set_mark_price(prices_manager):
     prices_manager.set_mark_price(10)
     assert prices_manager.mark_price == 10
-    assert prices_manager.mark_price_set_time == backtesting_exchange_manager.exchange.get_exchange_current_time()
-    assert prices_manager.valid_price_received_event.is_set()
+    if not os.getenv('CYTHON_IGNORE'):
```
Comment (GuillaumeDSM) [inline tests/data_manager/test_prices_manager.py:47]: why is it not working in cython anymore ?

File: tests/data_manager/test_recent_trades_manager.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        recent_trades_manager.liquidations.append(2)
+        assert recent_trades_manager.recent_trades
+        assert recent_trades_manager.liquidations
+        recent_trades_manager._reset_recent_trades()
+        assert not recent_trades_manager.recent_trades
+        assert not recent_trades_manager.liquidations
+        with pytest.raises(IndexError):
+            recent_trades_manager.recent_trades[0]
+        with pytest.raises(IndexError):
+            recent_trades_manager.liquidations[0]
```
Comment (GuillaumeDSM) [inline tests/data_manager/test_recent_trades_manager.py:127]: 👍

File: tests/data_manager/test_prices_manager.py (hunk 6 above)
Comment (Herklos) [reply tests/data_manager/test_prices_manager.py:47]: I don't know how it worked but `prices_manager.exchange_manager` can't be accessed when testing with cython compilation.

**Pull request merged**